pkg/aws/throttle/throttler_test.go

```diff
@@ -2,16 +2,99 @@ package throttle
 
 import (
 	"context"
+	"github.com/aws/aws-sdk-go/aws/client/metadata"
 	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/appmesh"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/time/rate"
 	"net/http"
+	"regexp"
 	"sync"
 	"sync/atomic"
 	"testing"
 	"time"
 )
 
+func Test_NewThrottler(t *testing.T) {
+	config := ServiceOperationsThrottleConfig{
+		value: map[string][]throttleConfig{
+			appmesh.ServiceID: {
```

Comment:
Thanks for adding the tests. 

nit: would be nice to use services/service names associated with this project. AppMesh looks a bit odd here given the controller has nothing to do with it.